flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -665,6 +674,19 @@ private void executeInvoke() throws Exception {
         afterInvoke();
     }
 
+    private void throughputCalculationSetup() {
+        systemTimerService.registerTimer(
+                systemTimerService.getCurrentProcessingTime()
```

Comment:
`systemTimerService.scheduleAtFixedRate()`?

Reply:
I thought about that but if the delay happens it can lead to the situation when `scheduleAtFixedRate` submit several throughput calculations which don't make sense. It is better to submit the calculation only when the last calculation was finished.